src/main/java/net/sf/jabref/logic/importer/fileformat/ModsImporter.java

```diff
@@ -0,0 +1,427 @@
+                String type = identifier.getType();
+                if ("citekey".equals(type) && !entry.getCiteKeyOptional().isPresent()) {
+                    entry.setCiteKey(identifier.getValue());
+                } else if (!"local".equals(type) && !"citekey".equals(type)) {
+                    //put all identifiers (doi, issn, isbn,...) except of local and citekey
+                    putIfValueNotNull(fields, identifier.getType(), identifier.getValue());
+                }
+            } else if (groupElement instanceof TitleInfoDefinition) {
+                TitleInfoDefinition titleInfo = (TitleInfoDefinition) groupElement;
+                List<Object> titleOrSubTitleOrPartNumber = titleInfo.getTitleOrSubTitleOrPartNumber();
+                for (Object object : titleOrSubTitleOrPartNumber) {
+                    if (object instanceof JAXBElement) {
+                        JAXBElement<StringPlusLanguage> element = (JAXBElement<StringPlusLanguage>) object;
+                        if ("title".equals(element.getName().getLocalPart())) {
+                            StringPlusLanguage title = element.getValue();
+                            fields.put(FieldName.TITLE, title.getValue());
+                        }
+                    }
+                }
+            }
+        }
+
+        //The element subject can appear more than one time, that's why the keywords has to be put here
+        putIfListIsNotEmpty(fields, keywords, FieldName.KEYWORDS, KEYWORD_SEPARATOR);
+        //same goes for authors and notes
+        putIfListIsNotEmpty(fields, authors, FieldName.AUTHOR, " and ");
+        putIfListIsNotEmpty(fields, notes, FieldName.NOTE, ", ");
+
+    }
+
+    private void parseGeographicInformation(Map<String, String> fields,
+            HierarchicalGeographicDefinition hierarchichalGeographic) {
+        List<JAXBElement<? extends StringPlusLanguage>> areaOrContinentOrCountry = hierarchichalGeographic
+                .getExtraTerrestrialAreaOrContinentOrCountry();
+        for (JAXBElement<? extends StringPlusLanguage> element : areaOrContinentOrCountry) {
+            String localName = element.getName().getLocalPart();
+            if ("city".equals(localName)) {
+                StringPlusLanguage city = element.getValue();
+                putIfValueNotNull(fields, "city", city.getValue());
+            } else if ("country".equals(localName)) {
+                StringPlusLanguage country = element.getValue();
+                putIfValueNotNull(fields, "country", country.getValue());
+            }
+        }
+    }
+
+    private void parseLocationAndUrl(Map<String, String> fields, LocationDefinition location) {
+        List<String> locations = new ArrayList<>();
+        List<PhysicalLocationDefinition> physicalLocations = location.getPhysicalLocation();
+        for (PhysicalLocationDefinition physicalLocation : physicalLocations) {
```

Comment:
Can't this be replaced by `location.getPhysicalLocation().stream().map(loc -> loc.getValue()).forEach(locations::add);`